test/client/metadataTransfer.test.ts

```diff
@@ -170,6 +169,27 @@ describe('MetadataTransfer', () => {
       expect(listenerStub.callCount).to.equal(1);
     });
 
+    it('should wait for polling function to return before queing another', async () => {
```

Comment:
This sets the polling frequency to 20 ms but the checkStatus function returns in 50 ms.  With the old PollingClient the callOrder array would be ['firstCall1', 'secondCall1', 'firstCall2'] but with the fix it's ['firstCall1', 'firstCall2', 'secondCall1'].